ambry-frontend/src/main/java/com/github/ambry/frontend/GetStatsReportHandler.java

```diff
@@ -0,0 +1,163 @@
+      // Start the callback chain by performing request security processing.
+      securityService.processRequest(restRequest, securityProcessRequestCallback());
+    }
+
+    /**
+     * After {@link SecurityService#processRequest} finishes, call {@link SecurityService#postProcessRequest} to perform
+     * request time security checks that rely on the request being fully parsed and any additional arguments set.
+     * @return a {@link Callback} to be used with {@link SecurityService#processRequest}.
+     */
+    private Callback<Void> securityProcessRequestCallback() {
+      return buildCallback(metrics.getStatsReportSecurityProcessRequestMetrics,
+          securityCheckResult -> securityService.postProcessRequest(restRequest, securityPostProcessRequestCallback()),
+          restRequest.getUri(), logger, finalCallback);
+    }
+
+    /**
+     * After {@link SecurityService#postProcessRequest} finishes, fetch stats report from {@link AccountStatsStore} and construct
+     * a json object to send back to {@link RestResponseChannel}.
+     * @return a {@link Callback} to be used with {@link SecurityService#postProcessRequest}.
+     */
+    private Callback<Void> securityPostProcessRequestCallback() {
+      return buildCallback(metrics.getStatsReportSecurityPostProcessRequestMetrics, securityCheckResult -> {
+        String clusterName =
+            RestUtils.getHeader(restRequest.getArgs(), RestUtils.Headers.GET_STATS_REPORT_CLUSTER, true);
+        String reportType = RestUtils.getHeader(restRequest.getArgs(), RestUtils.Headers.GET_STATS_REPORT_TYPE, true);
+        StatsReportType statsReportType;
+        try {
+          statsReportType = StatsReportType.valueOf(reportType);
+        } catch (Exception e) {
+          throw new RestServiceException(reportType + " is not a valid StatsReportType",
+              RestServiceErrorCode.BadRequest);
+        }
+        StatsSnapshot snapshot;
+        synchronized (accountStatsStore) {
+          try {
+            switch (statsReportType) {
+              case ACCOUNT_REPORT:
+                snapshot = accountStatsStore.queryAggregatedAccountStatsByClusterName(clusterName);
+                break;
+              case PARTITION_CLASS_REPORT:
+                snapshot = accountStatsStore.queryAggregatedPartitionClassStatsByClusterName(clusterName);
+                break;
+              default:
+                throw new RestServiceException("StatsReportType " + statsReportType + "not supported",
+                    RestServiceErrorCode.BadRequest);
+            }
+          } finally {
+            accountStatsStore.closeConnection();
+          }
+        }
+        if (snapshot == null) {
+          throw new RestServiceException("StatsReport not found for clusterName " + clusterName,
+              RestServiceErrorCode.NotFound);
+        }
+        try {
+          byte[] jsonBytes = mapper.writeValueAsBytes(snapshot);
```

Comment:
An alternative is to convert it to JSONObject and leverage `static ReadableStreamChannel serializeJsonToChannel(JSONObject jsonObject)` method in FrontendUtils.java

Reply:
that's unnecessary. ObjectMapper is more elegant.